### PR TITLE
Clear out `MACH_O_TYPE` for unit test bundles that use static frameworks.

### PR DESCRIPTION
**Pull request by dnkoutso**
closes https://github.com/CocoaPods/CocoaPods/issues/8031

### CODE REVIEW

File: lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb [hunk 1]
```diff
@@ -285,6 +285,9 @@ def add_test_targets
                 # requires frameworks. For tests we always use the test target name as the product name
                 # irrelevant to whether we use frameworks or not.
                 configuration.build_settings['PRODUCT_NAME'] = name
+                # target_installer sets 'MACH_O_TYPE' for static frameworks ensure this does not propagate
+                # to test target.
+                configuration.build_settings.delete('MACH_O_TYPE')
```
Comment (dnkoutso) [inline lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb]: I really dislike that super class sets a bunch of these and we have to delete them but I won't refactor it for 1.6.0.

Comment (amorde) [reply lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb]: agree this isn't ideal

**Comment by dnkoutso** [comment]
This bug actually exists in 1.5.3 too.

**Comment by amorde** [comment]
Yeah looks like the bug was reported while using 1.5.3

**Pull request merged**